src/API/public.js

```diff
@@ -0,0 +1,36 @@
+import axios from 'axios';
+
+/*
+    class to make public api calls token can be null
+    private.js uses this method for authenticated calls.
+*/
+export default class Request {
+	constructor(token = null) {
+		this.token = token;
+        this.headers = {
+            //"X-CSRFToken": cookie.load("csrftoken"),
+            "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+            "cache-control": "no-cache",
+            "Authorization": token,
+            }
+    }
+
+    delete = async (url, bodyParamters={}) => {
+        let response = await axios.delete(url, bodyParamters, this.headers);
+        return response.data;
+    }
+
+    post = async (url, bodyParamters={}) => {
+        let response = await axios.post(url, bodyParamters, this.headers);
+        return response.data 
+    }
+
+    get = async (url, bodyParamters={}) => {
+        let response = await axios.get(url, bodyParamters, this.headers);
+        return response.data
+    }
+    patch = async (url, bodyParamters={}) => {
+        let response = await axios.patch(url, bodyParamters, this.headers);
+        return response.data
+    }
+}
```

Comment:
yo you can create instances of axios and setup pre-request interceptors


```js

      const redditApi = axios.create({
        baseURL: 'https://reddit.com',
        headers: { 'Authorization': `Bearer ${token}` },
        timeout: 10000, // default timeout
      });

      redditApi.interceptors.request.use((config: any) => {
        config.headers.dynamic = Date.now();
        return config;
      });


 // somewhere in the code after signin, we have a token
    redditApi.defaults.headers.common['Authorization'] = `Bearer ${token}`;

```

Reply:
we shouldnt need to manage `js-cookies` or `query-string`,  you can then use:

```
redditApi.get('/user/username', {
  params: {
    sort: 'new'
  }
})
```
```

Reply:
@ilovett 

Thats a much better solution to use intercepters as apposed to the other method. I knew intercepters existed but never really used them before.

do you think its a better approach if axios should be creating upon class init which is called from a function in API/index.js or just return an instance of redditAPI and locally make requests? 

my concerns with passing an instance of redditAPI is how to store it and pass it to components. without redux would we have to prop drill the instance to different components or would it just be something like this?


src/API/index.js

    import axios from 'axios';
    const instance = axios.create();

    export default instance;

in src/app.js

    import reditAPI from '/src/API/'
    redditApi.get()

Reply:
@Slurpgoose with `react-query` you can either make a custom hook or a reusable hook, i think its fine to not need a wrapper for a long time.

```jsx

const Component = () => {
  
  // anonymously in the component
  const query = useQuery('nameOfQuery', async () => {
     const users = redditApi.get('/users');
     return users.data;
  });

   // OR import a shared hook
   const users = useGetRedditUser(userId);

   return query.status === 'loading' ? <Spinner /> : query.data ? <Component data={query.data} /> : null;
}

const useGetRedditUser = (userId: string) => {
  // this shows how to pass a variable and use it in the callback
  return useQuery(['nameOfQuery', { userId }], async ({ queryKey: [, { userId ] }) => {
     const users = redditApi.get(`/users/${userId}`);

     // destructure response
     return users.data;
  });
}
```

While I think its a good idea to abstract or name all the api calls (CRUDL classes) but I think its a bit complicated at this time...

Reply:
I never knew about react-query. thanks for the explanation always cool to learn something new! I'll close this PR.